qemu/tests/qemu_guest_agent.py

```diff
@@ -461,24 +461,34 @@ def gagent_check_get_vcpus(self, test, params, env):
         :param params: Dictionary with the test parameters
         :param env: Dictionary with test environment.
         """
-        error_context.context("Check can-offline field of guest agent.", logging.info)
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+
+        error_context.context("Check can-offline field of guest agent.",
+                              logging.info)
         vcpus_info = self.gagent.get_vcpus()
         cpu_num_qga = len(vcpus_info)
         for vcpu in vcpus_info:
             if params.get("os_type") == "linux":
-                if vcpu["logical-id"] == 0 and vcpu["can-offline"] is True:
-                    test.fail("The first logical vcpu can't be offline.")
+                if vcpu["logical-id"] == 0:
+                    vcpu_can_offline_qga = vcpu["can-offline"]
+                    cmd = "find /sys/devices/system/cpu/cpu0/ -name online"
+                    if session.cmd_output(cmd):
+                        vcpu_can_offline_guest = True
+                    else:
+                        vcpu_can_offline_guest = False
+                    if vcpu_can_offline_qga != vcpu_can_offline_guest:
+                        test.fail("The first logical vcpu's can-offline field"
+                                  " isn't aligned with it's in guest.")
                 if vcpu["logical-id"] != 0 and vcpu["can-offline"] is False:
```

Comment:
So Amd host has `online` file under the directory `/sys/devices/system/cpu/cpux/` for other logic cpu?

Reply:
yes.